### ISSUE TITLE
Chart shrinks with angular ui layout and bootstrap tab

### ISSUE THREAD

**Issue by dipakbhimani** [krispo/angular-nvd3#324]
Hi,

I have used angular ndv3 with the combination of 
1. Angular ui-layout ( https://github.com/angular-ui/ui-layout), I would require vertical split in my page
2. Angular UI Bootstrap (https://github.com/angular-ui/bootstrap)

With above combination, whenever graph rendered on page, it gets shrinks. If I resize windows it works perfectly. It seems very similar to https://github.com/krispo/angular-nvd3/issues/85.

I am quite new to angular world and angular-nvd3. Appreciate quick help..!!!

Example here - 
http://plnkr.co/edit/lJCMSvnDAGstex9oxyJH?p=preview

**Comment by jmsims2** [comment]
I've been experiencing the same issue in an ionic app using ng-show.  Thus far I've been trying to update the chart when I change the tab, but for some reason I can't get the api attribute to work.  In addition, when I call $scope.$apply() I get a stack error that says the apply is already in progress.  +1 for any help!

**Comment by samimaliks** [comment]
Yeah Guys I am also facing the same problem in ionic framework. This will only happen in my case  when I open any ionic.modal or popup. It shrinks ..I thinks die to visibility none. rather than display none.

Here is my stack question.

http://stackoverflow.com/questions/35307383/conflict-of-ionicmodal-with-angularnvd3-based-on-d3

**Comment by soumya2080** [comment]
assign a ng-click event to the ui-tab so that it'll append the data to graph at that time. And for `$scope.$apply()` write it like this way

```
if ($scope.$root.$$phase != '$apply' && $scope.$root.$$phase != '$digest') {
          $scope.$apply();
}
```

So that you can get rid of `$apply() ` already in progress error.

**Comment by samimaliks** [comment]
Hi Guys I solved this issue by changing width="100%" to width="100vw" in angular nvd3 directive.

like this line:

scope.svg.attr('width', '100%').style({width: '100%'});
changed to:

scope.svg.attr('width', '100%').style({width: '100vw'});
I used the following SO link which helps you to know the difference between 100% and 100vw:

Difference between Width:100% and width:100vw?
